protocol/messenger.go

```diff
@@ -151,11 +151,12 @@ type peerStatus struct {
 }
 type mailserverCycle struct {
 	sync.RWMutex
-	activeMailserver *enode.Node // For usage with wakuV1
-	activeStoreNode  *peer.ID    // For usage with wakuV2
-	peers            map[string]peerStatus
-	events           chan *p2p.PeerEvent
-	subscription     event.Subscription
+	activeMailserver          *enode.Node // For usage with wakuV1
+	activeStoreNode           *peer.ID    // For usage with wakuV2
+	peers                     map[string]peerStatus
+	events                    chan *p2p.PeerEvent
+	subscription              event.Subscription
+	availabilitySubscriptions map[chan struct{}]chan string
```

Comment:
This was my attempt to create a way to subscribe inside status-go to whether a mailserver is available.

There's probably a better way to do this, please let me know.